fastcrypto-zkp/src/conversions.rs

```diff
@@ -191,6 +195,171 @@ pub fn blst_g2_affine_to_bls_g2_affine(pt: &blst_p2_affine) -> BlsG2Affine {
     BlsG2Affine::new(ptx, pty, infinity)
 }
 
+/////////////////////////////////////////////////////////////
+// Zcash point encodings to Arkworks points and back       //
+/////////////////////////////////////////////////////////////
+
+// The standard for serialization of compressed G1, G2 points of BLS12-381
+// is not followed by Arkworks 0.3.0. This is a translation layer to allow
+// us to use (and receive) the standard serialization format.
+// See section 5.4.9.2 of https://zips.z.cash/protocol/protocol.pdf
+// and https://datatracker.ietf.org/doc/draft-irtf-cfrg-bls-signature/ Appendix A
+// for its choice as a standard.
+
+// A note on BLST: blst uses zcash point encoding formats across the board.
+
+fn bls_fq_to_zcash_bytes(field: &Fq) -> [u8; G1_COMPRESSED_SIZE] {
+    let mut result = [0u8; G1_COMPRESSED_SIZE];
+
+    let rep = field.into_repr();
+
+    result[0..8].copy_from_slice(&rep.0[5].to_be_bytes());
+    result[8..16].copy_from_slice(&rep.0[4].to_be_bytes());
+    result[16..24].copy_from_slice(&rep.0[3].to_be_bytes());
+    result[24..32].copy_from_slice(&rep.0[2].to_be_bytes());
+    result[32..40].copy_from_slice(&rep.0[1].to_be_bytes());
+    result[40..48].copy_from_slice(&rep.0[0].to_be_bytes());
+
+    result
+}
+
+fn bls_fq_from_zcash_bytes(bytes: &[u8; G1_COMPRESSED_SIZE]) -> Option<Fq> {
+    let mut tmp = BigInteger384([0, 0, 0, 0, 0, 0]);
```

Comment:
that value could be a const and reusable

Reply:
Yes, except this is actually not used as a value at all but as an initializer, which is exhaustively rewritten. 

The dominating cost is the allocation, and setting this as a const would actually not change. Further, the entire type `BigInteger384` is not used anywhere else, as this line shows:
https://github.com/MystenLabs/fastcrypto/pull/147/files#diff-b99b0ce42bc8ac2a0026f647de15eb9f02f0c7bcbf3b7ad6b7ff23d628041177R3

Reply:
maybe there is .zero()
https://github.com/arkworks-rs/algebra/blob/666e80dcfc0754f89580240c854fed2e86080449/ff/src/biginteger/mod.rs#L128

Reply:
I'll replace it with `default`, which makes more sense here (we don't care about the value).